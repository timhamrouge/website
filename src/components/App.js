import React, { Component } from 'react'
// import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';

// const App = ({message}) =><h1>Hello from {message}</h1>; 



class App extends Component {

    state = {
        name: '',
        message: "rob"
    };

render() {
    console.log(this.state)

    const { message } = this.state;

    return(
        <div>
        <h1>{message} is a fanny</h1>
        <form>
            <input type="text" name="fname" onChange={this.handleChange}/>
        <input type="submit" onClick={this.handleSubmit}></input>
        </form>
        </div>
    )
}

handleChange = () => {
    this.setState({ name: event.target.value })
    }

handleSubmit = (event) => {
    event.preventDefault();
    const { name } = this.state;
    this.setState({ message: name })
}

}

export default hot(module)(App);