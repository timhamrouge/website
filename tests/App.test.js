import React from 'react';
import renderer from 'react-test-renderer';
import App from 'App';

describe('App Component Test Suite', () => {
  it('Should render properly to DOM', () => {
    const wrapper = renderer.create();
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});