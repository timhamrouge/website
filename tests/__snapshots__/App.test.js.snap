// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Test Suite Should render properly to DOM 1`] = `null`;
